c7n/policy.py

```diff
@@ -87,18 +87,28 @@ def from_data(cls, data, options):
     def __add__(self, other):
         return self.__class__(self.policies + other.policies, self.options)
 
-    def filter(self, policy_name=None, resource_type=None):
+    def filter(self, policy_patterns=[], resource_types=[]):
         results = []
-        for policy in self.policies:
-            if resource_type:
-                if policy.resource_type != resource_type:
-                    continue
-            if policy_name:
-                if not fnmatch.fnmatch(policy.name, policy_name):
-                    continue
+        for policy in self.__filter_by_patterns(policy_patterns):
+            if resource_types and policy.resource_type not in resource_types:
+                continue
             results.append(policy)
         return PolicyCollection(results, self.options)
 
+    def __filter_by_patterns(self, patterns):
```

Comment:
let's just use a single underscore here for marking as private, the double under causes name mangling for references outside of the class which doesn't really add value imo.